### ISSUE TITLE
Error when running Nordic EPUB3 Validator

### ISSUE THREAD

**Issue by svenax** [nlbdev/pipeline#1]
Running `dp2 nordic-epub3-validate` gives the error below. Same result when using the GUI.

    Duplicate step type: letex:unzip

**Comment by josteinaj** [comment]
Hi @svenax. Thanks for the report.

Are you building this project ([nlbdev/pipeline](https://github.com/nlbdev/pipeline/)), or are you using the latest release of [nlbdev/nordic-epub3-dtbook-migrator](http://nlbdev.github.io/nordic-epub3-dtbook-migrator/)? The latter is what we use in production, and should work with [Pipeline 2 v1.9](https://github.com/daisy/pipeline-assembly/releases/tag/v1.9). It does not yet work with [Pipeline 2 v1.10](https://github.com/daisy/pipeline-assembly/releases/tag/v1.10.0), althought I've planned to get that working in the future.

[This repository](https://github.com/nlbdev/pipeline/) is Pipeline 2 v1.10 + NLB scripts + the Nordic EPUB3/DTBook Migrator scripts. However, this repository isn't fully working yet. I might make a "nordic" and a "nlb" version of Pipeline 2 once I get this repository working, so that the installation process for you and anyone else wanting to use the nordic migrator is a bit easier.

But for now, I suggest using Pipeline 2 v1.9 and install the nordic migrator as described in the [nordic migrator user guide](http://nlbdev.github.io/nordic-epub3-dtbook-migrator/guide/index.html#/1) (download from https://dl.dropboxusercontent.com/u/6370535/nordic-epub3-dtbook-migrator/index.html).

**Comment by svenax** [comment]
Thanks. I was building *this* project. I'll try to use *nordic-epub3-dtbook-migrator* instead.

**Comment by svenax** [comment]
Hmm. Downloaded Pipeline v1.9 from here https://github.com/daisy/pipeline-assembly/releases/download/v1.9/pipeline2-1.9-webui_mac.zip, nordic stuff from here https://dl.dropboxusercontent.com/u/6370535/nordic-epub3-dtbook-migrator/builds/nordic-epub3-dtbook-migrator-1.1.0-build_314-commit_13e61e4.jar and dependencies from here https://dl.dropboxusercontent.com/u/6370535/nordic-epub3-dtbook-migrator/epubcheck-adapter-with-dependencies/dependencies-1.8.1.zip. 

Running `./cli/dp2 nordic-epub3-validate` still gives the same error

    (1)[ERROR]	Duplicate step type: letex:unzip

**Comment by josteinaj** [comment]
Hi.

I tried it now in a clean environment and it worked fine. Using docker:

```bash
➜  docker run --rm -it java:7 bash
root@9c963ed02386:/# apt update && apt install wget
root@9c963ed02386:/# cd
root@9c963ed02386:~# wget https://github.com/daisy/pipeline-assembly/releases/download/v1.9/pipeline2-1.9-webui_linux.zip
root@9c963ed02386:~# wget https://dl.dropboxusercontent.com/u/6370535/nordic-epub3-dtbook-migrator/builds/nordic-epub3-dtbook-migrator-1.1.0-build_314-commit_13e61e4.jar
root@9c963ed02386:~# wget https://dl.dropboxusercontent.com/u/6370535/nordic-epub3-dtbook-migrator/epubcheck-adapter-with-dependencies/dependencies-1.8.1.zip
root@9c963ed02386:~# unzip pipeline2-1.9-webui_linux.zip 
root@9c963ed02386:~# unzip dependencies-1.8.1.zip 
root@9c963ed02386:~# mv modules/* daisy-pipeline/modules/
root@9c963ed02386:~# rm modules/ -r
root@9c963ed02386:~# mv nordic-epub3-dtbook-migrator-1.1.0-build_314-commit_13e61e4.jar daisy-pipeline/modules/
root@9c963ed02386:~# cd daisy-pipeline/
root@9c963ed02386:~/daisy-pipeline# cli/dp2 help
Launching the pipeline webservice...
The webservice is UP!

Usage dp2 [GLOBAL_OPTIONS] command [COMMAND_OPTIONS] [PARAMS]


Script commands:

        dtbook-to-html                       Transforms DTBook XML into HTML. [v1.0.2]
        nordic-epub3-validate                Validates an EPUB3 publication according to the nordic markup guidelines. [v1.0.0]
        daisy3-to-epub3                      Transforms a DAISY 3 publication into an EPUB 3 publication. [v1.1.1]
        daisy202-validator                   Validates a DAISY 2.02 fileset. [v1.0.0]
        nordic-dtbook-to-html                Transforms a DTBook document into an single HTML file according to the nordic markup guidelines. [v1.0.0]
        nordic-epub3-to-html                 Transforms an EPUB3 publication into HTML5 according to the nordic markup guidelines. [v1.0.0]
        dtbook-to-zedai                      Transforms DTBook XML into ZedAI XML. [v1.0.2]
        nordic-dtbook-to-epub3               Transforms a DTBook document into an EPUB3 publication according to the nordic markup guidelines. [v1.0.0]
        zedai-to-html                        Transforms ZedAI XML (ANSI/NISO Z39.98-2012 Authoring and Interchange) into HTML. [v1.0.2]
        nimas-fileset-validator              Validate a NIMAS Fileset. Supports inclusion of MathML. [v1.0.2]
        dtbook-to-pef                        Transforms a DTBook (DAISY 3 XML) document into a PEF. [v1.1.1]
        dtbook-to-epub3                      Converts multiple dtbooks to epub3 format [v1.2.0]
        daisy202-to-epub3                    Transforms a DAISY 2.02 publication into an EPUB3 publication. [v1.1.1]
        nordic-html-validate                 Validates a single-document HTML publication according to the nordic markup guidelines. [v1.0.0]
        dtbook-to-daisy3                     Converts multiple dtbooks to daisy 3 format [v2.0.0]
        zedai-to-epub3                       Transforms a ZedAI (DAISY 4 XML) document into an EPUB 3 publication. [v1.2.0]
        html-to-epub3                        Transforms (X)HTML documents into an EPUB 3 publication. [v1.1.1]
        zedai-to-pef                         Transforms a ZedAI (DAISY 4 XML) document into a PEF. [v1.1.1]
        nordic-dtbook-validate               Validates an dtbook publication according to the nordic markup guidelines. [v1.0.0]
        nordic-epub3-asciimath-to-mathml     Creates MathML for all HTML elements with the "asciimath" class. (experimental) [v1.0.0]
        epub3-to-daisy202                    Transforms an EPUB3 publication into DAISY 2.02. [v1.0.0]
        nordic-epub3-to-dtbook               Transforms an EPUB3 publication into DTBook according to the nordic markup guidelines. [v1.0.0]
        nordic-html-to-dtbook                Transforms a HTML5 document into a DTBook according to the nordic markup guidelines. [v1.0.0]
        nordic-html-to-epub3                 Transforms a HTML document into an EPUB3 publication according to the nordic markup guidelines. [v1.0.0]
        dtbook-validator                     Validates DTBook documents. Supports inclusion of MathML. [v1.0.2]
        

General commands:

        status                               Returns the status of the job with id JOB_ID
        delete                               Removes a job from the pipeline
        results                              Stores the results from a job
        jobs                                 Returns the list of jobs present in the server
        log                                  Stores the results from a job
        queue                                Shows the execution queue and the job's priorities. 
        moveup                               Moves the job up the execution queue
        movedown                             Moves the job down the execution queue
        halt                                 Stops the webservice
        version                              Prints the version and authentication information
        sizes                                Prits the total size or a detailed list of job data stored in the server
        


List of global options:                 dp2 help -g
List of admin commands:                 dp2 help -a
Detailed help for a single command:     dp2 help COMMAND
root@9c963ed02386:~/daisy-pipeline# cli/dp2 nordic-epub3-validate
Error:
	option/switch --x-epub is mandatory for command nordic-epub3-validate
root@9c963ed02386:~/daisy-pipeline# wget https://github.com/nlbdev/nordic-epub3-dtbook-migrator/blob/master/src/test/xprocspec/C00000.epub?raw=true
root@9c963ed02386:~/daisy-pipeline# mv C00000.epub\?raw\=true C00000.epub
root@9c963ed02386:~/daisy-pipeline# cli/dp2 nordic-epub3-validate --x-epub C00000.epub --output /tmp
Job 35c8c620-ede1-4979-8abc-e8d7be005dce sent to the server
(1)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/package.opf from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/package.opf
(2)[INFO]	trying to read xml declaration from C00000-01-cover.xhtml
(3)[INFO]	xml declaration from C00000-01-cover.xhtml is ?xml version="1.0" encoding="UTF-8"?
(4)[INFO]	trying to read doctype declaration from C00000-01-cover.xhtml
(5)[INFO]	doctype declaration from C00000-01-cover.xhtml is: !DOCTYPE html
(6)[INFO]	trying to read xml declaration from C00000-03-frontmatter.xhtml
(7)[INFO]	xml declaration from C00000-03-frontmatter.xhtml is ?xml version="1.0" encoding="UTF-8"?
(8)[INFO]	trying to read doctype declaration from C00000-03-frontmatter.xhtml
(9)[INFO]	doctype declaration from C00000-03-frontmatter.xhtml is: !DOCTYPE html
(10)[INFO]	trying to read xml declaration from C00000-02-toc.xhtml
(11)[INFO]	xml declaration from C00000-02-toc.xhtml is ?xml version="1.0" encoding="UTF-8"?
(12)[INFO]	trying to read doctype declaration from C00000-02-toc.xhtml
(13)[INFO]	doctype declaration from C00000-02-toc.xhtml is: !DOCTYPE html
(14)[INFO]	trying to read xml declaration from C00000-04-chapter.xhtml
(15)[INFO]	xml declaration from C00000-04-chapter.xhtml is ?xml version="1.0" encoding="UTF-8"?
(16)[INFO]	trying to read doctype declaration from C00000-04-chapter.xhtml
(17)[INFO]	doctype declaration from C00000-04-chapter.xhtml is: !DOCTYPE html
(18)[INFO]	trying to read xml declaration from C00000-05-chapter.xhtml
(19)[INFO]	xml declaration from C00000-05-chapter.xhtml is ?xml version="1.0" encoding="UTF-8"?
(20)[INFO]	trying to read doctype declaration from C00000-05-chapter.xhtml
(21)[INFO]	doctype declaration from C00000-05-chapter.xhtml is: !DOCTYPE html
(22)[INFO]	trying to read xml declaration from C00000-06-chapter.xhtml
(23)[INFO]	xml declaration from C00000-06-chapter.xhtml is ?xml version="1.0" encoding="UTF-8"?
(24)[INFO]	trying to read doctype declaration from C00000-06-chapter.xhtml
(25)[INFO]	doctype declaration from C00000-06-chapter.xhtml is: !DOCTYPE html
(26)[INFO]	trying to read xml declaration from C00000-07-rearnotes.xhtml
(27)[INFO]	xml declaration from C00000-07-rearnotes.xhtml is ?xml version="1.0" encoding="UTF-8"?
(28)[INFO]	trying to read doctype declaration from C00000-07-rearnotes.xhtml
(29)[INFO]	doctype declaration from C00000-07-rearnotes.xhtml is: !DOCTYPE html
(30)[INFO]	trying to read xml declaration from C00000-08-chapter.xhtml
(31)[INFO]	xml declaration from C00000-08-chapter.xhtml is ?xml version="1.0" encoding="UTF-8"?
(32)[INFO]	trying to read doctype declaration from C00000-08-chapter.xhtml
(33)[INFO]	doctype declaration from C00000-08-chapter.xhtml is: !DOCTYPE html
(34)[INFO]	trying to read xml declaration from C00000-09-conclusion.xhtml
(35)[INFO]	xml declaration from C00000-09-conclusion.xhtml is ?xml version="1.0" encoding="UTF-8"?
(36)[INFO]	trying to read doctype declaration from C00000-09-conclusion.xhtml
(37)[INFO]	doctype declaration from C00000-09-conclusion.xhtml is: !DOCTYPE html
(38)[INFO]	trying to read xml declaration from C00000-10-footnotes.xhtml
(39)[INFO]	xml declaration from C00000-10-footnotes.xhtml is ?xml version="1.0" encoding="UTF-8"?
(40)[INFO]	trying to read doctype declaration from C00000-10-footnotes.xhtml
(41)[INFO]	doctype declaration from C00000-10-footnotes.xhtml is: !DOCTYPE html
(42)[INFO]	trying to read xml declaration from nav.ncx
(43)[INFO]	xml declaration from nav.ncx is ?xml version="1.0" encoding="UTF-8"?
(44)[INFO]	skipping doctype check for non-HTML document: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.ncx
(45)[INFO]	trying to read xml declaration from nav.xhtml
(46)[INFO]	xml declaration from nav.xhtml is ?xml version="1.0" encoding="UTF-8"?
(47)[INFO]	trying to read doctype declaration from nav.xhtml
(48)[INFO]	doctype declaration from nav.xhtml is: !DOCTYPE html
(49)[INFO]	trying to read xml declaration from package.opf
(50)[INFO]	xml declaration from package.opf is ?xml version="1.0" encoding="UTF-8"?
(51)[INFO]	skipping doctype check for non-HTML document: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/package.opf
(52)[INFO]	trying to read xml declaration from container.xml
(53)[INFO]	xml declaration from container.xml is ?xml version="1.0" encoding="UTF-8"?
(54)[INFO]	skipping doctype check for non-HTML document: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/META-INF/container.xml
(55)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml
(56)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml
(57)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml
(58)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml
(59)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml
(60)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml
(61)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml
(62)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml
(63)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml
(64)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml
(65)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.xhtml
(66)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.ncx from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.ncx
(67)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml
(68)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml
(69)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml
(70)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml
(71)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml
(72)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml
(73)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml
(74)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml
(75)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml
(76)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml
(77)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-01-cover.xhtml
(78)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-03-frontmatter.xhtml
(79)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-02-toc.xhtml
(80)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-04-chapter.xhtml
(81)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-05-chapter.xhtml
(82)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-06-chapter.xhtml
(83)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-07-rearnotes.xhtml
(84)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-08-chapter.xhtml
(85)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-09-conclusion.xhtml
(86)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/C00000-10-footnotes.xhtml
(87)[INFO]	loading file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.xhtml from disk: file:/root/daisy-pipeline/data/jobs/35c8c620-ede1-4979-8abc-e8d7be005dce/output/temp-dir/validate/EPUB/nav.xhtml

Results stored
The job has been deleted from the server
Job finished with status: DONE
root@9c963ed02386:~/daisy-pipeline# 
```

I see you're using OSX. I haven't tried that but it works on both Windows and Linux so I doubt there should be any trouble with OSX. There's nothing OS-specific in the nordic migrator source code.

Are you sure you're using Pipeline v1.9? If you already had Pipeline v1.10 running then the CLI won't shut it down. Maybe try `./cli/dp2 halt` first and see if that helps?

**Comment by josteinaj** [comment]
To see which version of Pipeline is currently running you can go to http://localhost:8181/ws/alive where you'll get something like:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<alive authentication="false" localfs="true" version="1.9" xmlns="http://www.daisy.org/ns/pipeline/data"/>
```

**Comment by svenax** [comment]
Running the exact same commands as you did in a clean docker image works fine. It looks like I could use `docker-nlb-base` for provisioning, right?. I first tried `snaekobbi/system` from docker-hub, but realised that it doesn’t have the same modules.

**Comment by josteinaj** [comment]
Ah, yes, [docker-nlb-base](https://github.com/nlbdev/docker-nlb-base) should work. We're not currently using it here but the nordic scripts should work there. I've been wanting to use docker more but haven't gotten around to it. If you want to use docker-nlb-base then you should probably lock its version (`docker-nlb-base:v1.0.0`) in case I change it in the future.

**Comment by josteinaj** [comment]
Hmm, I tried docker-nlb-base on my computer now but didn't get it working. Don't know why it didn't work, I'll have to update it soon. Meanwhile I'm sure you could make a Dockerfile based on the commands I posted above if you want to.

**Comment by svenax** [comment]
OK. Trying it `dp2` just hangs. I'll use a Dockerfile of my own for now.

**Comment by josteinaj** [comment]
This is now fixed and merged to the `nlb` branch as of 495e22928d397ad4c698aaf25d7fd39b19ec4ccc.
